pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -210,6 +213,15 @@ func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(ctx context.Context, o
 			moa.logger.Error("failed to apply Alertmanager config for org", "org", orgID, "id", dbConfig.ID, "error", err)
 			continue
 		}
+
+		// Mark config as successfully applied if needed.
+		if dbConfig.SuccessfullyAppliedAt == 0 {
+			moa.logger.Debug("Marking Alertmanger configuration as successfully applied", "org", orgID, "id", dbConfig.ID)
+			if err := moa.configStore.MarkAlertmanagerConfigurationAsSuccessful(ctx, dbConfig.ID); err != nil {
```

Comment:
That's another good example for naming. 

Here the context is high, as you already operate on the config store, so the information that it's an alertmanger configuration is not really needed in the function name. So a naming like `moa.configStore.MarkApplied` would be way more readable.

Reply:
For the record, the `*DBstore` struct implements an interface that has 7 methods, all containing "AlertmanagerConfiguration" in their name.
To keep consistency, I named this method `MarkAlertmanagerConfigurationAsSuccessful`, which has the exact same amount of characters as other methods like `SaveAlertmanagerConfigurationWithCallback`.
The name `configStore` is used in the multiorg AM, but the name of the actual struct doesn't imply that its methods act on AM configurations.